### ISSUE TITLE
"Not found" error is returned once data is cleared and typed again

### ISSUE THREAD

**Issue by alirezastack** [gmerabishvili/angular-ng-autocomplete#131]
#### Bug description
When I type a text (with bounceTime) it gets data from API and displays it in ng-autocomplete with no issue. Now when I clear data in input and type a text like `A50` it returns data based on the given search keyword from API (I console.log() the API response to make sure data is returned). But I get `Not Found` error. Now when I press backspace and remove a character it again receives data from API, but this time it shows data in ng-autocomplete list!

I have reproduced this many times.

#### Expected result
Data should be returned every time I enter a text, clear data and again type something.

#### Actual result
`Not Found` error is displayed once I enter a text, clear the entered search keyword and enter a new search keyword again.

#### Steps to reproduce
 1. Type a text (it gets response from API and fill the autocomplete)
 2. Now clear the text by select the whole search keyword and enter backspace (Not character by character).
 3. Enter a new search keyword. `Not Found` should be displayed.

#### Your Environment

* Version used: 2.0.2
* Browser Name and version: Chrome 87.0
* Operating System and version: Mac OS X Big Sur

#### Code

**search.component.html**

```
<ng-autocomplete
                [data]="supplierSelector"
                [searchKeyword]="supplierSearchKeyword"
                (selected)='selectEvent($event)'
                (inputChanged)='onChangeSearch($event)'
                (inputFocused)='onFocused($event)'
                [itemTemplate]="supplierTemplate"
                [notFoundTemplate]="notFoundTemplate" required
                [formControl]="getFormControl('supplier')"
              >
</ng-autocomplete>
```

**search.component.ts**

```
productsSelectorChange = new Subject<any>();
modelChanged: Subject<string> = new Subject<string>();
subscription: Subscription;
productsSelector: { name: string, _id: string }[] = [];

constructor(private _formBuilder: FormBuilder,
              private shoppingListService: ShoppingListService,
              private productsService: ProductsService,
              private toastr: ToastrService
  ) {
    this.productsService.searchSubject.pipe(map((response) => {
      // clear old data in ng-autocomplete
      this.productsSelector.splice(0, this.productsSelector.length);
      this.products = response['data'];
      this.products.forEach(product => {
        const product_last_category = product.categories.pop().category;
        this.productsSelector.push(
          {name: product.name + ' | ' + product_last_category, _id: product._id}
        );
      });
      this.productsSelectorChange.next(this.productsSelector);
      this.isLoading = false;
    })).subscribe(products => {
    }, error => {
      toastr.error(error, 'Prouduct Fetch Error');
    }, () => {
    });
    this.subscription = this.modelChanged.pipe(debounceTime(1000)).subscribe((modelName) => {
      this.productsService.search(modelName);
    });
  }

onChangeSearch(val: string) {
    this.isLoading = true;
    this.modelChanged.next(val);
  }

ngOnDestroy(): void {
    this.subscription.unsubscribe();
}
```

**Comment by gmerabishvili** [comment]
@alirezastack As I guess you're filtering on server side, right?  Try use "isLoading" property and let me know the result. Thank you.

**Comment by alirezastack** [comment]
@gmerabishvili I have added *[isLoading]* and implemented it, but it seems that it just displays a loading icon when data is being fetched from API. 

I logged API response and I saw that data is returned from API but ng-autocomplete is not filled with returned data and loading icon keeps displaying (in the scenario I explained in the main question).

This is weird that just by removing a single character it calls *onChangeSearch* and it then displays data from API correctly.
